File: engine/src/main/java/ru/vertuos/engine/hedgehog/listener/HedgehogPropertyChangeListener.java
package ru.vertuos.engine.hedgehog.listener;

import ru.vertuos.engine.hedgehog.Hedgehog;
import ru.vertuos.engine.world.object.DynamicGameObject;

public interface HedgehogPropertyChangeListener {

    void onPositionChange(float newPosX, float newPosY);

    void onLinearVelocityChange(float newVelX, float newVelY);

    void onAccelerationChange(float newAccX, float newAccY);

    void onStateChange(Hedgehog.State newState);

    void onHpChange(float newHp);

    void onLivesCountChange(int newLivesCount);

    void onRingsCountChange(int newRingsCount);

    void onOnGroundChange(boolean isOnGroundNow);

    void onChangeDirection(DynamicGameObject.Direction dir);
}

File: engine/src/main/java/ru/vertuos/engine/hedgehog/listener/HedgehogPropertyChangeListener.kt
package ru.vertuos.engine.hedgehog.listener

import ru.vertuos.engine.hedgehog.Hedgehog
import ru.vertuos.engine.world.obj.DynamicGameObject.Direction

interface HedgehogPropertyChangeListener {

    fun onPositionChange(newPosX: Float, newPosY: Float)

    fun onLinearVelocityChange(newVelX: Float, newVelY: Float)

    fun onAccelerationChange(newAccX: Float, newAccY: Float)

    fun onStateChange(newState: Hedgehog.State)

    fun onHpChange(newHp: Float)

    fun onLivesCountChange(newLivesCount: Int)

    fun onRingsCountChange(newRingsCount: Int)

    fun onOnGroundChange(isOnGroundNow: Boolean)

    fun onChangeDirection(dir: Direction)
}
